x-pack/plugins/security_solution/public/transforms/utils/is_filter_query_compatible.ts

```diff
@@ -5,17 +5,50 @@
  * 2.0.
  */
 
+import { isEqual } from 'lodash/fp';
 import { ESQuery } from '../../../common/typed_json';
 
-export const isFilterQueryCompatible = (filterQuery: ESQuery | string | undefined) => {
+/**
+ * Array of query compatible objects which are at the moment all
+ * simple empty or match all based objects
+ */
+const queryCompatibleStrings: ESQuery[] = [
+  {
+    bool: {
+      must: [],
+      filter: [{ match_all: {} }],
+      should: [],
+      must_not: [],
+    },
+  },
+  {
+    bool: {
+      must: [],
+      filter: [],
+      should: [],
+      must_not: [],
+    },
+  },
+];
+
+/**
+ * Returns true if the filter query matches against one of the compatible strings, otherwise
+ * false. Right now we only check if the filter query is empty, or a match all in order to activate
+ * the transform.
+ * @param filterQuery The filterQuery to check against and return true if it matches, otherwise false.
+ * @returns true if the filter is compatible, otherwise false.
+ */
+export const isFilterQueryCompatible = (filterQuery: ESQuery | string | undefined): boolean => {
   if (filterQuery === undefined) {
     return true;
   } else if (typeof filterQuery === 'string') {
-    return (
-      filterQuery === '{"bool":{"must":[],"filter":[{"match_all":{}}],"should":[],"must_not":[]}}'
-    );
+    try {
+      const filterQueryObject = JSON.parse(filterQuery);
```

Comment:
This `filterQuery` will only ever be the filters from the main page-level KQL bar, correct? No need to worry about excessively large filters here?

Reply:
That is correct.